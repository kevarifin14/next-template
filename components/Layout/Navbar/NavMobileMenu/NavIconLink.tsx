import Link from 'next/link';

type NavIconLinkProps = {
  href: string,
  title: string,
  icon?: any,
}

export default function NavIconLink({ href, title, icon }: NavIconLinkProps) {
  return (
    <Link href={href}>
      <a className="-m-3 flex items-center p-3 rounded-lg hover:bg-light-200 no-underline">

        <div className="flex-shrink-0 flex items-center justify-center h-10 w-10 rounded-md bg-primary-500 text-white sm:h-12 sm:w-12">
          {icon({ className: 'text-white h-5 w-5' })}
        </div>

        <div className="ml-3 text-base font-medium text-gray-900">
          {title}
        </div>

      </a>
    </Link>
  );
}
