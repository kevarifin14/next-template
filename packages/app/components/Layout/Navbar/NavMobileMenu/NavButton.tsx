import { ReactNode } from 'react';

type NavButtonProps = {
  children: ReactNode,
  onClick: () => void,
};

export default function NavButton({ onClick, children }: NavButtonProps) {
  return (
    <button
      type="button"
      className="bg-light rounded-md p-2 inline-flex items-center justify-center text-light-500 hover:text-light-900 hover:bg-light-200 focus:outline-none"
      onClick={onClick}
    >
      {children}
    </button>
  );
}
