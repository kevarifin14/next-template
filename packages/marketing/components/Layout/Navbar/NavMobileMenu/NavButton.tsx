import { ReactNode } from 'react';

type NavButtonProps = {
  children: ReactNode,
  onClick: () => void,
};

export default function NavButton({ onClick, children }: NavButtonProps) {
  return (
    <button
      type="button"
      className="prose dark:prose-light dark:hover:bg-dark-light hover:bg-light-dark rounded-md p-2 inline-flex items-center justify-center focus:outline-none"
      onClick={onClick}
    >
      {children}
    </button>
  );
}
