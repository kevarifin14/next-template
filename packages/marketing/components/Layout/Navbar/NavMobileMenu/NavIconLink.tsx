import Link from 'next/link';

type NavIconLinkProps = {
  href: string,
  title: string,
  icon?: any,
};

export default function NavIconLink({ title, href, ...props }: NavIconLinkProps) {
  return (
    <Link href={href}>
      <a className="-m-3 flex items-center p-3 rounded-lg no-underline prose hover:bg-light-dark dark:prose-light dark:hover:bg-dark-500">

        <div className="flex-shrink-0 flex items-center justify-center h-10 w-10 rounded-md bg-primary sm:h-12 sm:w-12">
          <props.icon className="h-5 w-5 text-white" />
        </div>

        <p className="ml-3 text-base font-medium m-0 dark:text-gray-50">
          {title}
        </p>

      </a>
    </Link>
  );
}
