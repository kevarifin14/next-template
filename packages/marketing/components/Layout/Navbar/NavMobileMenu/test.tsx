import { act, fireEvent, render } from '@testing-library/react';
import { HiInbox } from 'react-icons/hi';

import NavMobileMenu from '.';

describe('NavMobileMenu', () => {
  const renderNavMobileMenu = (props = {}) => (
    render(
      <NavMobileMenu
        links={[
          {
            title: 'Packages',
            href: '/packages',
            icon: HiInbox,
          },
        ]}
        {...props}
      />,
    )
  );

  it('matches snapshot', () => {
    const { baseElement } = renderNavMobileMenu();
    expect(baseElement).toMatchSnapshot();
  });

  it('opens the menu', () => {
    const { getByRole, getAllByRole, queryByText } = renderNavMobileMenu();

    const openButton = getByRole('button');

    act(() => {
      fireEvent.click(openButton);
    });

    expect(queryByText('Packages')).toBeTruthy();

    const [, closeButton] = getAllByRole('button');

    act(() => {
      fireEvent.click(closeButton);
    });
  });
});
