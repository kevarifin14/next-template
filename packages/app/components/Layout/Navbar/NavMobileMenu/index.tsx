import { Transition } from '@headlessui/react';
import { MenuIcon, XIcon } from '@heroicons/react/outline';
import { useRouter } from 'next/router';
import { useState } from 'react';

import Button from 'components/Button';
import LogoImage from 'components/LogoImage';

import NavButton from './NavButton';
import NavIconLink from './NavIconLink';

export default function NavMobileMenu({ links }) {
  const [isOpen, setIsOpen] = useState(false);
  const router = useRouter();

  return (
    <>

      <div className="md:hidden">
        <NavButton onClick={() => setIsOpen(true)}>
          <MenuIcon className="h-6 w-6" />
        </NavButton>
      </div>

      <Transition
        show={isOpen}
        className="absolute z-30 top-0 inset-x-0 transition transform origin-top-right md:hidden"
        enter="duration-200 ease-out"
        enterFrom="opacity-0 scale-95"
        enterTo="opacity-100 scale-100"
        leave="duration-100 ease-in"
        leaveFrom="opacity-100 scale-100"
        leaveTo="opacity-0 scale-95"
      >
        <div className="rounded-lg shadow-lg bg-light divide-y-2 divide-gray-50">
          <div className="pt-5 pb-6 px-5 sm:pb-8">
            <div className="flex items-center justify-between">

              <div className="h-8 w-8 relative">
                <LogoImage />
              </div>

              <div className="-mr-2">
                <NavButton onClick={() => setIsOpen(false)}>
                  <XIcon className="h-6 w-6" />
                </NavButton>
              </div>

            </div>

            <div className="mt-6 sm:mt-8">
              <nav className="space-y-2">

                {links.map((props) => <NavIconLink key={props.title} {...props} />)}

                <Button type="primary" block onClick={() => router.push('/signin')}>
                  Sign Up
                </Button>

              </nav>
            </div>

          </div>
        </div>
      </Transition>
    </>
  );
}
