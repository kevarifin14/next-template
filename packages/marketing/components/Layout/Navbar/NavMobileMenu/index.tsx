import { Transition } from '@headlessui/react';
import { useRouter } from 'next/router';
import { useState } from 'react';
import { HiMenu, HiX } from 'react-icons/hi';
import { Button } from 'shared';

import LogoImage from 'components/LogoImage';

import NavButton from './NavButton';
import NavIconLink from './NavIconLink';

export default function NavMobileMenu({ links }) {
  const [isOpen, setIsOpen] = useState(false);
  const router = useRouter();

  return (
    <>

      <div className="md:hidden">
        <NavButton onClick={() => setIsOpen(true)}>
          <HiMenu className="h-6 w-6" />
        </NavButton>
      </div>

      <Transition
        show={isOpen}
        className="absolute z-30 top-0 inset-x-0 transition transform origin-top-right md:hidden"
        enter="duration-200 ease-out"
        enterFrom="opacity-0 scale-95"
        enterTo="opacity-100 scale-100"
        leave="duration-100 ease-in"
        leaveFrom="opacity-100 scale-100"
        leaveTo="opacity-0 scale-95"
      >
        <div className="rounded-lg shadow-lg bg-light dark:bg-dark-light divide-y-2 divide-gray-50">
          <div className="p-6 space-y-6">
            <div className="flex items-center justify-between">

              <div className="h-8 w-8 relative">
                <LogoImage />
              </div>

              <div className="-mr-2">
                <NavButton onClick={() => setIsOpen(false)}>
                  <HiX className="h-6 w-6" />
                </NavButton>
              </div>

            </div>

            <nav className="space-y-2">

              {links.map((props) => <NavIconLink key={props.title} {...props} />)}

            </nav>

            <Button type="primary" block onClick={() => router.push('/signin')}>
              Sign Up
            </Button>

          </div>
        </div>
      </Transition>
    </>
  );
}
