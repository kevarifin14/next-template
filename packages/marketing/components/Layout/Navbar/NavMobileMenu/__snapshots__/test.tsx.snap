// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMobileMenu matches snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="md:hidden"
      >
        <button
          class="prose dark:prose-light dark:hover:bg-dark-light hover:bg-light-dark rounded-md p-2 inline-flex items-center justify-center focus:outline-none"
          type="button"
        >
          <svg
            class="h-6 w-6"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
